src/main/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -252,15 +252,14 @@ private List<ItemStack> calculateSecondaryOutputs()
      */
     private void calculateTools()
     {
-        final List<ItemStorage> cleaned = getCleanedInput();
-        for (int index = 0; index < cleaned.size(); index++)
+        for (int index = 0; index < this.cleanedInput.size(); index++)
         {
-            final ItemStorage item = cleaned.get(index);
+            final ItemStorage item = this.cleanedInput.get(index);
             for (ItemStack result : getSecondaryOutputs())
             {
                 if (ItemStackUtils.compareItemStacksIgnoreStackSize(item.getItemStack(), result, false, true) && result.isDamageableItem())
                 {
-                    cleaned.set(index, new ImmutableItemStorage(new ItemStorage(item.getItemStack(), item.getAmount(), true)));
+                    this.cleanedInput.set(index, new ImmutableItemStorage(new ItemStorage(item.getItemStack(), item.getAmount(), true)));
```

Comment:
Using `this` here and not on the other fields is a bit inconsistent.  I'd omit that.